packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -214,14 +248,23 @@ class AOTSnapshotter {
       '-Xlinker', '-rpath', '-Xlinker', '@executable_path/Frameworks',
       '-Xlinker', '-rpath', '-Xlinker', '@loader_path/Frameworks',
       '-install_name', '@rpath/App.framework/App',
+      if (bitcode) '-fembed-bitcode',
       '-o', appLib,
       assemblyO,
     ];
     final RunResult linkResult = await xcode.clang(linkArgs);
     if (linkResult.exitCode != 0) {
       printError('Failed to link AOT snapshot. Linker terminated with exit code ${compileResult.exitCode}');
+      return linkResult;
+    }
+    final RunResult dsymResult = await xcode.dsymutil(<String>[
```

Comment:
This ends up creating 3 App.framework.dSYM folders in the build/ directory. Unfortunately, it seems like this issue https://github.com/flutter/flutter/pull/22540 affects local xcode too. If spotlight finds these files, any attempt to desymbolicate locally will attach to one of these files. But since we don't know which binary architecture the developer ended up using, xcode gets confused. Can we put these transient files in .noindex folders too?